mlir/utils/performance/perfRunner.py

```diff
@@ -793,6 +822,127 @@ def benchmarkMIOpenWithMLIRKernels(configs, arch, filename, paths: Paths):
+                parts = command.split('|')  # Split the command using the "|" separator
+                return parts[0].strip()
+                # Stop processing lines after finding the first matching line
+    # Not a valid test
+    print("WARNING: cannot find RUN command in the test file")
+    return ""
+
+def getTestVector(file_path, paths:Paths):
+    firstCommand = findRunCommand(file_path)
+    if (not firstCommand):
+        return ""
+
+    if ("-migraphx-to-tosa" in firstCommand):
+        rocmlirOptCommand = [paths.mlir_paths.rocmlir_opt_path, '-migraphx-to-tosa', file_path]
+        rocmlirDriverCommand = [paths.mlir_paths.rocmlir_driver_path, '-host-pipeline', 'partition,highlevel', '-targets', getChip()]
+        # rocmlir-opt -migraphx-to-tosa ../mlir/test/fusion/e2e/resnet50/mixr-resnet-fusion-case-1.mlir
+        p1 = subprocess.Popen(rocmlirOptCommand, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        # pipe to rocmlir-driver -host-pipeline partition,highlevel -targets gfx90a
+        p2 = subprocess.Popen(rocmlirDriverCommand, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        p1.stdout.close()
+    else:
+        rocmlirDriverCommand = [paths.mlir_paths.rocmlir_driver_path, '-host-pipeline', 'partition,highlevel', '-targets', getChip(), file_path]
+        # rocmlir-driver -host-pipeline partition,highlevel -targets gfx90a
+        p2 = subprocess.Popen(rocmlirDriverCommand, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+ 
+    # pipe to rocmlir_gen --emit-tuning-key
+    tuningKey = subprocess.Popen([paths.mlir_paths.rocmlir_gen_path, '--emit-tuning-key', '-'], stdin=p2.stdout,
+                          stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    p2.stdout.close()
+    output, _ = tuningKey.communicate()
+    result = output.decode('utf-8').strip().split('\t')
+    return result[1]
+
+def benchmarkFusionKernels(test_dir, paths: Paths, arch, tuningDb: MaybeTuningDb):
+    TABLE_COLUMNS = reportUtils.FUSION_TEST_COL
+    allData = []
+
+    chip = GFX_CHIP_RE.search(arch).group(0)
+    for filename in os.listdir(test_dir):
+        if filename.endswith(".mlir"):
+            os.system("rm "+BENCHMARKING_RESULT_FILE_NAME)
+
+            file_path = os.path.join(test_dir, filename)
+            print("Profiling:", file_path)
+            testVector = getTestVector(file_path, paths)
+            if (not testVector):
+                continue
+
+            commandLine = testVector.split(sep=' ')
+            if (commandLine[0].startswith('conv')):
+                op = 'conv'
+                config = ConvConfiguration.fromCommandLine(commandLine, arch)
+            else:
+                op = 'gemm'
+                config = GemmConfiguration.fromCommandLine(commandLine, arch)
+
+            # find the best perf_config
+            bestPerf =""
+            if tuningDb:
+                if (arch, filename, testVector) in tuningDb:
```

Comment:
Reject putting file names in the tuning DB.

The tuning DB should, for now, just key off the gemm or convolution, ignoring the fusions.

Reply:
Some test cases have the same tuning keys and the tuning-driver gave different best configs. Are they just noise? Is it ok to choose any best config?

Reply:
Ok, so
1. How distinct were those perf numbers?
2. Which of those configs, if any, corresponded to the config for just that GEMM without any fusions?

Reply:
That is to say, we should investigate that further. You'll also want to make sure you've merged in the tuner fixes from #1030 and #1032

Reply:
1. The perf numbers are not distinct much. For one test case I've looked at, there is just one parameter off by 4. But I didn't check other test cases.
2. Don't know yet.
I haven't merged #1030 and #1032 though.

Reply:
> Ok, so
> 
> 1. How distinct were those perf numbers?
> 2. Which of those configs, if any, corresponded to the config for just that GEMM without any fusions?

None of them did.

Reply:
> 
> 1. How distinct were those perf numbers?
 The TFlops resulting from different perf configs were very close, with differences of less than 1%. I think that was simply normal variation between runs .

Reply:
> That is to say, we should investigate that further. You'll also want to make sure you've merged in the tuner fixes from #1030 and #1032

Nothing changed with PR#1030 and #1032.